crates/ibc/src/host_helpers.rs

```diff
@@ -0,0 +1,112 @@
+use core::time::Duration;
+
+use alloc::format;
+use alloc::string::{String, ToString};
+use ibc_proto::google::protobuf::Any;
+
+use crate::clients::ics07_tendermint::client_state::ClientState as TmClientState;
+use crate::core::ics02_client::client_state::ClientState;
+use crate::core::ics03_connection::context::ConnectionReader;
+use crate::core::ics03_connection::error::Error;
+use crate::core::ics23_commitment::specs::ProofSpecs;
+use crate::core::ics24_host::identifier::ChainId;
+
+use tendermint::trust_threshold::TrustThresholdFraction as TendermintTrustThresholdFraction;
+
+pub trait TmValidateSelfClientContext {
```

Comment:
> If so, we can remove the ctx parameter as self would be the host itself.

That could indeed, if we added `ConnectionReader` as a super trait. The current approach gives the flexibility for hosts to implement `TmValidateSelfClientContext` on a different struct; not sure if this would be needed though?

Reply:
> Since this is public API, it would be nice to add some doc comments here.

[354f9fc](https://github.com/cosmos/ibc-rs/pull/207/commits/354f9fc855abdd1d6be6aedc229bd0c244b98a17)

Reply:
> That could indeed, if we added ConnectionReader as a super trait. The current approach gives the flexibility for hosts to implement TmValidateSelfClientContext on a different struct; not sure if this would be needed though?

Ah, will [apply this suggestion](https://github.com/cosmos/ibc-rs/pull/207#discussion_r1012623420) instead.